julien/liferay-portal
# LPS-114186 Migrate AUI tree-view in asset-taglib to React

Hey @wincent ,

This PR is follow up on

julien#284
julien#285
mateomustapic#9
mateomustapic#10
and
https://github.com/wincent/liferay-portal/pull/313

This PR is a work on migration of AUI tree-view to React in asset-taglib module.
The old aui tree-view implementation had completely different logic than React TreeView requires, so I Eudaldo created new Java classes which can now be used in new implementation.

I used the `<react:component>` tag and included React component in the `jsp` where old tree-view was located, and I additionally cleaned up unnecessary jsp and `AUI` code which was used in old implementation.
Additionally, Julien made modifications to react component for passing `categories` and `vocabularies` to `AssetCategoriesNavigationTreeView` component.

After Eudaldo's final changes in https://github.com/wincent/liferay-portal/pull/313 I modified JS to:
- Fix bug with 'ghost' icons in `TreeView` component 
- Fix categories navigate `url`
- Fix error when clicking on a `vocabulary` and redirect only if `category` selected

To test these changes :

Navigate to Categorization / Categories
Create some categories in Vocabularies list for test
Create a widget page (Site Builder/Pages new Blank Page)
Add the portlet categories navigation to a page
Now the tree view should be visible on created page.

If you need anything else to test this, please tell me.

I am sending screenshot of `TreeView` with fixed "missing icons" bug.
